agent/rpc/peering/replication.go

```diff
@@ -265,6 +269,16 @@ func (s *Service) handleUpdateService(
 			req.Service = svcSnap.Service
 			if err := s.Backend.Apply().CatalogRegister(&req); err != nil {
 				return fmt.Errorf("failed to register service: %w", err)
+			} else {
+				if !servicePresentInStored(req.Service, storedInstances) {
```

Comment:
Doing this in the loop will overcount the services. We would do the counting here if we wanted to measure the number of service *instances*.

The same thing goes for the delete.

I think something like the following at the very end of this method would work:

```
if len(storedInstances) > 0 && len(pbNodes.GetNodes()) < 1 {
   // decrement the service count as it was removed
} else if len(storedInstances) < 1 && len(pbNodes.GetNodes()) > 0 {
   // increment the service count as no instances previously existed but they do now.
}
```